### PR TITLE
Adds /api when necessary

### PR DESCRIPTION
**Pull request by aliaandersen**
If needed, will add /api to a api url that has /Contrast/ or /Contrast

### CODE REVIEW

**Comment by matt-** [comment]
Can we remove the .DS_Store files from this?

**Comment by cixel** [comment]
May help to add a `.gitignore` to prevent random files like this from being re-added.

File: src/main/java/com/contrastsecurity/sdk/ContrastSDK.java [hunk 1]
```diff
@@ -617,6 +617,6 @@ public void setReadTimeout(final int timeout) {
     private static final int SERVER_ERROR = 500;
 
     private static final String DEFAULT_API_URL = "https://app.contrastsecurity.com/Contrast/api";
-    private static final String LOCALHOST_API_URL = "http://localhost:19080/Contrast/api";
+    private static final String LOCALHOST_API_URL = "http://localhost:19090/Contrast/api";
```
Comment (gmoore) [inline src/main/java/com/contrastsecurity/sdk/ContrastSDK.java]: I Think this needs to stay at 19080

File: src/main/java/com/contrastsecurity/utils/ContrastSDKUtils.java [hunk 2]
```diff
@@ -35,6 +35,16 @@ public static void validateUrl(String url) throws IllegalArgumentException {
         }
     }
 
+    public static String ensureApi(String url){
+        if(url.endsWith("/Contrast")){
+            url = url + "/api";
+        }
+        else if( url.endsWith("/Contrast/")){
```
Comment (gmoore) [inline src/main/java/com/contrastsecurity/utils/ContrastSDKUtils.java]: move the else up a line so it looks like `} else {`

File: src/test/java/com/contrastsecurity/ContrastSDKTest.java [hunk 3]
```diff
@@ -120,5 +119,19 @@ public void negativeTimeoutsAreNotSetTest() throws IOException {
...
+    public void testEnsureApi() throws IOException {
+        final String expectedApi = "http://localhost:19080/Contrast/api";
+        final String ensureUrlOne = ContrastSDKUtils.ensureApi("http://localhost:19080/Contrast/");
+        final String ensureUrlTwo = ContrastSDKUtils.ensureApi("http://localhost:19080/Contrast");
+
+        assertEquals(ensureUrlOne, expectedApi);
+        assertEquals(ensureUrlTwo, expectedApi);
+
+        final String ensureUrlIncorrect = ContrastSDKUtils.ensureApi("http://localhost:19080/");
+        assertNotEquals(ensureUrlIncorrect, expectedApi);
```
Comment (gmoore) [inline src/test/java/com/contrastsecurity/ContrastSDKTest.java]: Change this assertion to assert that the string is unchanged. Our method changes the string if in ends in Contrast, but if it doesn't end in Contrast, then the string should be returned unchanged.

File: src/main/java/com/contrastsecurity/sdk/ContrastSDK.java [hunk 4]
```diff
@@ -121,7 +121,10 @@ public ContrastSDK(String user, String serviceKey, String apiKey, String restApi
         this.apiKey = apiKey;
         this.restApiURL = restApiURL;
 
+        this.restApiURL = restApiURL;
```
Comment (gmoore) [inline src/main/java/com/contrastsecurity/sdk/ContrastSDK.java]: Dupe'd line :)

File: src/test/java/com/contrastsecurity/ContrastSDKTest.java [hunk 5]
```diff
@@ -120,5 +119,20 @@ public void negativeTimeoutsAreNotSetTest() throws IOException {
     	assertNotEquals(connectionTimeout, conn.getConnectTimeout());
     	assertNotEquals(readTimeout, conn.getReadTimeout());
     }
+
+    @Test
+    public void testEnsureApi() throws IOException {
```
Comment (gmoore) [inline src/test/java/com/contrastsecurity/ContrastSDKTest.java]: Good test cleanup. Let's add a few more test cases here: 

* Malformed URL
* Null URL
* Blank URL

But these in separate test methods

**Pull request merged**